pkg/slack/events/joblink/link.go

```diff
@@ -0,0 +1,389 @@
+package joblink
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gcsupload"
+	"k8s.io/test-infra/prow/pjutil"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+	gcsutil "k8s.io/test-infra/prow/pod-utils/gcs"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/jobconfig"
+	"github.com/openshift/ci-tools/pkg/rehearse"
+	"github.com/openshift/ci-tools/pkg/slack/events"
+)
+
+// job abstracts the types of Prow jobs. Only one field can be non-nil
+type job struct {
+	metadata   api.Metadata
+	presubmit  *config.Presubmit
+	postsubmit *config.Postsubmit
+	periodic   *config.Periodic
+}
+
+// JobGetter knows how to retrieve job details for a job by name
+type JobGetter interface {
+	JobForName(name string) *job
+}
+
+// NewJobGetter exposes a simplified interface from the Prow configuration
+func NewJobGetter(config config.Getter) JobGetter {
+	return &getter{
+		config: config,
+	}
+}
+
+type getter struct {
+	config config.Getter
+}
+
+func metadataFromConfig(orgRepo string, job config.JobBase, brancher config.Brancher) api.Metadata {
+	parts := strings.Split(orgRepo, "/")
+	if len(parts) != 2 {
+		return api.Metadata{}
+	}
+	return api.Metadata{
+		Org:     parts[0],
+		Repo:    parts[1],
+		Branch:  rehearse.BranchFromRegexes(brancher.Branches),
+		Variant: rehearse.VariantFromLabels(job.Labels),
+	}
+}
+
+func (g *getter) JobForName(name string) *job {
+	var found job
+
+	for orgRepo, presubmits := range g.config().PresubmitsStatic {
```

Comment:
I don't think the name is enough, I think we have the same name on different branches

Reply:
Not for generated jobs. I'm pretty sure name is unique in our config.